spec/ics-25-handler-interface/README.md

```diff
@@ -0,0 +1,314 @@
+
+`ConnectionInfo` contains metadata about & state of an existing connection.
+
+```golang
+type ConnectionInfo struct {
+  ConnectionOptions options
+  ConnectionState   state
+}
+```
+
+`initConnection` tries to create a new connection with the provided options, failing if the client is not found or the options are invalid.
+
+```coffeescript
+function initConnection(identifier, desiredVersion, desiredCounterpartyIdentifier, lightClientIdentifier) -> Maybe<err>
+```
+
+`tryConnection` tries to initialize a connection based on an initialization attempt on another chain (part one of the three-way connection handshake).
+
+```coffeescript
+function tryConnection(desiredIdentifier, counterpartyIdentifier, desiredVersion, counterpartyLightClientIdentifier, lightClientIdentifier, proofInit) -> Maybe<err>
+```
+
+`ackConnection` acknowledges a connection in progress on another chain (part two of the three-way connection handshake).
+
+```coffeescript
+function ackConnection(identifier, agreedVersion, proofTry)
+```
+
+`confirmConnection` finalizes a connection (part three of the three-way connection handshake).
+
+```coffeescript
+function confirmConnection(identifier, proofAck)
+```
+
+`queryConnection` queries an existing connection by known identifier, returning the associated metadata if found.
+
+```coffeescript
+function queryConnection(string id) -> Maybe<ConnectionInfo>
+```
+
+`initCloseConnection` initiates the graceful connection closing process. It will fail if there are any open channels using the connection.
+
+```coffeescript
+function initCloseConnection(string id) -> Maybe<Err>
+```
+
+`tryCloseConnection` continues the graceful connection closing process. It will fail if there are any open channels using the connection.
+
+```coffeescript
+function tryCloseConnection(identifier, proofInit) -> Maybe<Err>
+```
+
+`ackCloseConnection` finalizes the graceful connection closing process.
+
+```coffeescript
+function ackCloseConnection(identifier, proofTry) -> Maybe<Err>
+```
+
+`equivocationCloseConnection` force-closes a connection upon discovery of an equivocation. The equivocation must first be submitted to the associated client.
+
+```coffeescript
+function equivocationCloseConnection(identifier) -> Maybe<Err>
+```
+
+Implementations of `initConnection`, `tryConnection`, `ackConnection`, `confirmConnection`, `queryConnection`, `initCloseConnection`, `tryCloseConnection`, `ackCloseConnection`, and `equivocationCloseConnection` are defined in ICS 3.
+
+### Channels
+
+By default, channels are owned by the creating module, meaning only the creating module can inspect, close, or send on the channel.
+
+`ChannelOptions` contains all the parameter choices required to create a new channel.
+
+```golang
+type ChannelOptions struct {
+  string          connectionIdentifier
+  bool            ordered
+}
+```
+
+`ChannelInfo` contains metadata about an existing channel.
+
+```golang
+type ChannelInfo struct {
+  ChannelOptions options
+}
+```
```

Comment:
There is no need for a `channelIdentifier`?

Reply:
Ah yes, indeed there is; added.